source/agora/consensus/EnrollmentManager.d

```diff
@@ -659,53 +693,77 @@ public class EnrollmentManager
     ***************************************************************************/
 
     private Height getNextRevealHeight () @safe nothrow
+    {
+        return Height(loadIntegerData("next_reveal_height", ulong.max));
+    }
+
+    /***************************************************************************
+
+        Set the next block height to reveal a pre-image
+
+        Params:
+            height = the next block height to reveal a pre-image
+
+    ***************************************************************************/
+
+    private void setNextRevealHeight (Height height) @safe nothrow
+    {
+        saveIntegerData("next_reveal_height", height);
+        this.next_reveal_height = height;
+    }
+
+    /***************************************************************************
+
+        Save data of integer type into the table
+
+        Params:
+            name = the name for the data
+            value = the value of the data
+
+    ***************************************************************************/
+
+    private void saveIntegerData (string name, size_t value) @trusted nothrow
     {
         try
         {
-            return () @trusted {
-                auto results = this.db.execute("SELECT val FROM node_enroll_data " ~
-                    "WHERE key = ?", "next_reveal_height");
-                if (!results.empty)
-                    return Height(results.oneValue!(ulong));
-                return Height(ulong.max);
-            }();
+            this.db.execute("REPLACE into node_enroll_data " ~
+                "(key, val) VALUES (?, ?)", name, value);
         }
         catch (Exception ex)
         {
             log.error("ManagedDatabase operation error {}", ex);
         }
-        return Height(ulong.max);
     }
 
     /***************************************************************************
 
-        Set the next block height to reveal a pre-image
+        Load data of integer type from the table
 
         Params:
-            height = the next block height to reveal a pre-image
+            name = the name for the data
+            default_value = the default value used when the data is missing
+
+        Returns:
+            the value of the data
 
     ***************************************************************************/
 
-    private void setNextRevealHeight (Height height) @safe nothrow
+    private size_t loadIntegerData (string name, size_t default_value = 0)
```

Comment:
If you change it to `T loadIntegerData (T)(string name)` you won't have to do any casting. The same goes for `saveIntegerData`, writing `void saveIntegerData (T)(string name, T value)`.